core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -1589,6 +1589,67 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
         sortsAs("[IS NULL($0)]"));
   }
 
+  @Test public void testPullUpPredicatesFromUnion() {
+    final RelMetadataQuery mq = RelMetadataQuery.instance();
+    RelNode rel = null;
+    rel = convertSql(""
+        + "select empno from emp where empno=1\n"
+        + "union all\n"
+        + "select empno from emp where empno=1");
+    assertThat(mq.getPulledUpPredicates(rel).pulledUpPredicates,
+        sortsAs("[=($0, 1)]"));
+
+    rel = convertSql(""
+        + "select empno, deptno from emp where empno=1 or deptno=2\n"
+        + "union all\n"
+        + "select empno, deptno from emp where empno=3 or deptno=4");
+    assertThat(mq.getPulledUpPredicates(rel).pulledUpPredicates,
```

Comment:
The rel of this sql is
```
LogicalUnion(all=[true])
  LogicalProject(EMPNO=[$0], DEPTNO=[$7])
    LogicalFilter(condition=[OR(=($0, 1), =($7, 2))])
      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
  LogicalProject(EMPNO=[$0], DEPTNO=[$7])
    LogicalFilter(condition=[OR(=($0, 3), =($7, 4))])
      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
```

after pull up, it will become
```
 LogicalFilter(condition=[OR(=($0, 1), =($1, 2), =($0, 3), =($1, 4))])
     LogicalUnion(all=[true])
         LogicalProject(EMPNO=[$0], DEPTNO=[$7])
             LogicalTableScan(table=[[CATALOG, SALES, EMP]])
         LogicalProject(EMPNO=[$0], DEPTNO=[$7])
             LogicalTableScan(table=[[CATALOG, SALES, EMP]])
```
Does not seem right?

Reply:
No, this change is not doing pull-up. RelMdPredicates is for analyzing constraints/characteristics of expression.
`[OR(=($0, 1), =($1, 2), =($0, 3), =($1, 4))]` says all rows in this `Union` satisfies such a condition